deeppavlov/dataset_readers/odqa_reader.py

```diff
@@ -0,0 +1,212 @@
+    `Wiki Extractor <https://github.com/attardi/wikiextractor>`_ files.
+
+    """
+
+    def read(self, data_path: Union[Path, str], db_url: Optional[str] = None, *args, **kwargs) -> None:
+        """Build a SQLite database from provided files, download SQLite database from a provided URL, or do nothing.
+
+        Args:
+            data_path: a directory/file with texts to create a database from
+            db_url: path to a database url
+
+        Returns:
+            None
+
+        """
+        logger.info('Reading files...')
+        save_path = expand_path(kwargs['save_path'])
+        if save_path.exists() and Path(save_path.parent / '.done').exists():
+            return
+
+        if db_url:
+            if Path(save_path).is_file():
+                download_dir = save_path.parent
+            else:
+                download_dir = save_path
+            logger.info(f'Downloading database from {db_url} to {download_dir}')
+            download(download_dir, db_url, force_download=False)
+            return
+
+        self._build_db(save_path, kwargs['dataset_format'], expand_path(data_path))
+
+    def iter_files(self, path: Union[Path, str]) -> Generator[Path, Any, Any]:
+        """Iterate over folder with files or a single file and generate file paths.
+
+        Args:
+            path: path to a folder or a file
+
+        Raises:
+            RuntimeError if the provided `path` doesn't exist
+
+        Yields:
+            file paths one by one
+
+        Returns:
+            None
+
+        """
+        path = Path(path)
+        if path.is_file():
+            yield path
+        elif path.is_dir():
+            for item in path.iterdir():
+                yield from self.iter_files(item)
+        else:
+            raise RuntimeError("Path doesn't exist: {}".format(path))
+
+    def _build_db(self, save_path: Union[Path, str], dataset_format: str, data_path: Union[Path, str],
+                  num_workers: int = 8) -> None:
+        """Build a SQLite database in parallel and save it to a pointed path.
+
+        Args:
+            save_path: a path where the ready database should be saved
+            dataset_format: a data format, should be selected from ['txt', 'json', 'wiki']
+            data_path: path to a folder/file from which to build a database
+            num_workers: a number of workers for parallel database building
+
+        Raises:
+            sqlite3.OperationalError if `save_path` doesn't exist.
+            RuntimeError if dataset_format is not in ['txt', 'json', 'wiki']
+
+        Returns:
+            None
+
+        """
+        done_path = Path(save_path.parent / '.done')
+
+        if Path(save_path).exists():
+            Path(save_path).unlink()
+        if done_path.exists():
+            done_path.unlink()
+
+        logger.info('Building the database...')
+
+        try:
+            conn = sqlite3.connect(str(save_path))
```

Comment:
I think we should not punish users for not wanting to create directories by hand.
`save_path.parent.mkdir(parents=True, exist_ok=True)`

Reply:
I think we should not punish users for making typos either. However, will do as accepted by voting.